journal/src/main/java/io/camunda/zeebe/journal/file/JournalSegmentDescriptor.java

```diff
@@ -52,32 +53,36 @@
   private final int encodedLength;
 
   private final SegmentDescriptorDecoder segmentDescriptorDecoder = new SegmentDescriptorDecoder();
-  private final MessageHeaderDecoder messageHeaderDecoder = new MessageHeaderDecoder();
+  private final MessageHeaderDecoder headerDecoder = new MessageHeaderDecoder();
   private final MutableDirectBuffer directBuffer = new UnsafeBuffer();
 
   private final SegmentDescriptorEncoder segmentDescriptorEncoder = new SegmentDescriptorEncoder();
-  private final MessageHeaderEncoder messageHeaderEncoder = new MessageHeaderEncoder();
+  private final MessageHeaderEncoder headerEncoder = new MessageHeaderEncoder();
 
   public JournalSegmentDescriptor(final ByteBuffer buffer) {
     directBuffer.wrap(buffer);
 
     /* The first byte of the buffer contains the version at which the descriptor is written.
      Currently we have only one version, so we can ignore it. We can use this version in future
     when needed. Note that we can add fields to SBE schema of the descriptor without changing this version. */
-    messageHeaderDecoder.wrap(directBuffer, VERSION_LENGTH);
+    headerDecoder.wrap(directBuffer, VERSION_LENGTH);
+
+    if (headerDecoder.schemaId() != segmentDescriptorDecoder.sbeSchemaId()
+        || headerDecoder.templateId() != segmentDescriptorDecoder.sbeTemplateId()) {
+      throw new CorruptedLogException("Cannot read segment descriptor. Header does not match.");
```

Comment:
Add more context to the error message such as expected schemaId,templateId and read schemaId and templateId.